Automattic/jetpack
# Search: Use aliased Preact imports

#### Changes proposed in this Pull Request:
In preparation for using our Instant Search application within Gutenberg, we'll rely on using aliased Preact imports using our Webpack configuration. 

(In other words, our codebase will read as though it's importing React even though it's actually importing Preact.)

#### Jetpack product discussion
Necessary for our future customization efforts, see p1HpG7-c6N-p2.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Apply this change to your Jetpack site with a Search subscription.
* Ensure that the search application builds as expected.
* Ensure that the search interface works as expected.
